### PR TITLE
Add SWEBench-docker eval

### PR DESCRIPTION
**Pull request by xingyaoww**
Attempt to address: https://github.com/OpenDevin/OpenDevin/issues/2072

Integrating https://github.com/aorwall/SWE-bench-docker for more reliable SWE-Bench evaluation

### CODE REVIEW

**Comment by xingyaoww** [comment]
This should be ready to review! The new eval harness (our fork here: https://github.com/OpenDevin/SWE-bench-docker, with some improvements on patch applying), from my observation, improves our task success rate by roughly 0.5-1%.

**Review by neubig — APPROVED** [review]
Looks great, just a few comments! Happy to merge after these are resolved as you see fit.

File: evaluation/swe_bench/README.md [hunk 1]
```diff
@@ -127,6 +127,12 @@ If you want to evaluate existing results, you should first run this to clone exi
 git clone https://huggingface.co/spaces/OpenDevin/evaluation evaluation/evaluation_outputs
 ```
 
+To prepare for swe-bench evaluation, you should pull evaluation docker from [aorwall/SWE-bench-docker](https://github.com/aorwall/SWE-bench-docker) and download swe-bench data by running:
```
[the remark this answers is not in the document]
Comment (xingyaoww) [reply evaluation/swe_bench/README.md:130]: Done!

File: evaluation/swe_bench/scripts/docker/pull_all_eval_docker.sh [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+mkdir evaluation/swe_bench/eval_workspace
+pushd evaluation/swe_bench/eval_workspace
+git clone -b xw/od-fix https://github.com/OpenDevin/SWE-bench-docker.git
```
[the remark this answers is not in the document]
Comment (xingyaoww) [reply evaluation/swe_bench/scripts/docker/pull_all_eval_docker.sh]: Yep - i already merge that to main but forget to change this. Updated!

**Comment by xingyaoww** [comment]
@neubig done address those points! I've enabled auto-merge

**Pull request merged**